sentry/src/main/java/io/sentry/cache/EnvelopeCache.java

```diff
@@ -454,4 +437,29 @@ public boolean waitPreviousSessionFlush() {
   public void flushPreviousSession() {
     previousSessionLatch.countDown();
   }
+
+  public void movePreviousSession(
+      final @NotNull File currentSessionFile, final @NotNull File previousSessionFile) {
+    try (final @NotNull ISentryLifecycleToken ignored = sessionLock.acquire()) {
+      if (previousSessionFile.exists()) {
```

Comment:
I'm wondering if we should simply overwrite the file instead. If the file still exists, chances are that we've already sent that session information. On the other hand `currentSessionFile` was never flushed, so renaming it to `previousSessionFile` and later flushing that session feels more correct.

Reply:
yeah I've been torn between this approach and what you suggest. Basically, if the `previous_session` file still exists it means the previous process exited unexpectedly before we managed to finish the previous-previous session and flush it (due to a crash or any other process death). In that case that session still hasn't been properly finalized so we're anyway losing out on it. So the question is which session do we want to lose 😅 

I guess to properly fix it we need some better storage of sessions (like a ring buffer, not just previous + current), but that's probably for later. I will change it to your suggestion though 👍